server/src/main/java/org/eclipse/hono/server/MessageForwardingEndpoint.java

```diff
@@ -38,13 +40,24 @@
  */
 public abstract class MessageForwardingEndpoint extends BaseEndpoint {
 
+    protected CounterService              counterService;
```

Comment:
But then we still need to provide the null-implementation in case the particular spring implementation is not available, which bloats the code on different places. Also I tend to agree with https://en.wikipedia.org/wiki/Null_Object_pattern#Criticism, that it might not be the right solution to just avoid null-checks. A Null-conditional operator would be great.. but not available in pure Java. 
Do you really think we need to be prepared for non-spring environments?